From a038d6624eb0bb4479d11f5b2bf472714c200365 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Sun, 1 Sep 2024 23:00:42 +0200
Subject: [PATCH] Do not lose input in xfig -debug mode

Starting with xfig -debug, entering line drawing mode and clicking on
the dimension line attributes in the indicator panel (below the canvas)
would lose the input, one has to quit xfig from the window manager or by
signal.
---
 src/w_indpanel.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git src/w_indpanel.c src/w_indpanel.c
index 5ab33f6..9e4755b 100644
--- src/w_indpanel.c
+++ src/w_indpanel.c
@@ -2995,10 +2995,8 @@ void popup_dimline_panel(ind_sw_info *isw)
 	actions_added = True;
     }
 
-    if (appres.DEBUG)
-	XtPopup(choice_popup, XtGrabNone);	/* makes debugging easier */
-    else
-        XtPopup(choice_popup, XtGrabExclusive);
+    XtPopup(choice_popup, XtGrabExclusive);
+
     /* if the file message window is up add it to the grab */
     file_msg_add_grab();
     (void) XSetWMProtocols(tool_d, XtWindow(choice_popup), &wm_delete_window, 1);
-- 
2.35.3

